### ISSUE TITLE
Error message

### ISSUE THREAD

**Issue by tomivirkki** [vaadin/vaadin-text-field#26]
Specs by @jouni

It should be possible to provide an error message for the text-field, which gets announced by the screen reader, for example in the case of a validation error.

The error message should only be displayed when the text field is invalid.

**API proposal**:
```html
<vaadin-text-field error-message="This field is required" label="Name" required></vaadin-text-field>
```

**Implementation details**

Expose the error message as a new themable part, e.g. 
```html
# shadow root
  ...
  <div class="error" part="error-message">[[errorMessage]]</div>
  ...
```

The error message should be displayed under the field by default. No extra styling is needed (colors or anything else).

**Comment by Saulis** [comment]
- Let's take inspiration for the screen reader implementation of `<paper-input>`, which announces the error message nicely.
